### PR TITLE
Various changes to the API and small fixes for the odri_control_interface

### PR DESCRIPTION
**Pull request by jviereck**
This is a bunch of fixes and small changes to the odri_control_interface.

Still missing:

- [x] Throw exception when there is an error during calibration / when waiting for the robot to become available.
- [x] Update the C++ example to match more the python one

### CODE REVIEW

File: include/odri_control_interface/calibration.hpp [hunk 1]
```diff
@@ -48,12 +48,13 @@ class JointCalibrator
     double Kp_;
     double Kd_;
     double T_;
-    double dt_;
     double t_;
     bool go_to_zero_position_;
     int n_;
 
 public:
+    double dt;
```
Comment (MaximilienNaveau) [inline include/odri_control_interface/calibration.hpp]: missing trailing underscore

Comment (MaximilienNaveau) [reply include/odri_control_interface/calibration.hpp]: I think it's better in c++ to separate the internal data to the front end api.
If you need access to the dt simply add a getter.
If you need to set it, use a setter.

You can do both like those:
```
// write only
void dt(double in)
{
    dt_ = in;
}
// read only
const double& dt()
{
    return dt_;
}
// read/write
double& dt()
{
    return dt_;
}
```

Comment (MaximilienNaveau) [reply include/odri_control_interface/calibration.hpp]: ps: pinocchio use the last 2, I think it's fairly elegant and prevent potential mistake removing the trailing underscore.

File: src/robot.cpp [hunk 2]
```diff
@@ -200,6 +217,22 @@ void Robot::WaitUntilReady()
...
+            throw std::runtime_error("Error during Robot::WaitUntilReady().");
+        }
+    }
+}
+
+void Robot::Initialize()
+{
+    Start();
+    WaitUntilReady();
+    RunCalibration();
```
Comment (MaximilienNaveau) [inline src/robot.cpp:235]: So when the executable of the robot is launch, the calibration is run right away?
We can try.

Comment (MaximilienNaveau) [reply src/robot.cpp:235]: Is that compatible with the DGM way of running?

Comment (MaximilienNaveau) [reply src/robot.cpp:235]: I guess yes because Start, WaitUntilReady and RunCalibration are public anyway :thinking:

File: srcpy/bindings.cpp [hunk 3]
```diff
@@ -203,19 +203,26 @@ BOOST_PYTHON_MODULE(libodri_control_interface_pywrap)
         .value("negative", CalibrationMethod::NEGATIVE)
         .value("alternative", CalibrationMethod::ALTERNATIVE);
 
+    bool (Robot::*RunCalibration0)() = &Robot::RunCalibration;
```
Comment (MaximilienNaveau) [inline srcpy/bindings.cpp:206]: super nice trick! this makes the code more elegant below

File: include/odri_control_interface/calibration.hpp (hunk 1 above)
Comment (jviereck) [reply include/odri_control_interface/calibration.hpp]: Agreed, I will change it.

File: src/robot.cpp (hunk 2 above)
Comment (jviereck) [reply src/robot.cpp:235]: Yes. The public API is not changed. This is just a new convenience method to get the python example look a bit shorter (Thomas said a PD controller should take only 10 lines, and with this change it does as you don't need to call Start, WaitUntilReady, RunCalibration manyally, which would yield 12 lines ^^). This means our way of running the code to trigger the calibration manually is working still just fine (I checked it).

File: srcpy/bindings.cpp (hunk 3 above)
Comment (jviereck) [reply srcpy/bindings.cpp:206]: Thanks the boost python doc for it :D

**Pull request merged**